### PR TITLE
[improvement](spill) avoid occuping too much memory while spill build block during the hash join build phase

### PR DESCRIPTION
**Pull request by mrhhsg**
## Proposed changes

1. Use `uint32_t` as the type of partitioned index.
2. Spill the build block in small blocks each time.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by mrhhsg** [comment]
run buildall

File: be/src/pipeline/exec/partitioned_hash_join_sink_operator.cpp [hunk 1]
```diff
@@ -45,8 +45,21 @@ Status PartitionedHashJoinSinkLocalState::init(doris::RuntimeState* state,
 
 Status PartitionedHashJoinSinkLocalState::open(RuntimeState* state) {
     RETURN_IF_ERROR(PipelineXSpillSinkLocalState::open(state));
+    auto& p = _parent->cast<PartitionedHashJoinSinkOperatorX>();
```
Comment (yiguolei) [inline be/src/pipeline/exec/partitioned_hash_join_sink_operator.cpp:48]: 我们不一定落盘，但是在这里去register 可能会导致有一些overhead？

**Review by yiguolei — APPROVED** [review]
LGTM

**Review by jacktengg — APPROVED** [review]
LGTM

**Pull request merged**